src/test/java/cloud/localstack/awssdkv2/BasicFeaturesSDKV2Test.java

```diff
@@ -106,12 +105,34 @@ public void testSendSNSMessage() throws Exception {
     public void testGetSsmParameter() throws Exception {
         // Test integration of ssm parameter with LocalStack using SDK v2
 
-        final String topicName = "test-t-"+UUID.randomUUID().toString();
         final SsmAsyncClient clientSsm = TestUtils.getClientSSMAsyncV2();
-        CompletableFuture<PutParameterResponse> putParameterReponse = clientSsm.putParameter(PutParameterRequest.builder().name("testparameter").value("testvalue").build());
-        CompletableFuture<GetParameterResponse>  getParameterResponse = clientSsm.getParameter(GetParameterRequest.builder().name("testparameter").build());
+        clientSsm.putParameter(PutParameterRequest.builder().name("testparameter").value("testvalue").build());
```

Comment:
Thanks for adding this test and refactoring the code a bit @jellybeat !

Looks like we still have minor test errors in this test. I'm not too familiar with the Java SDK, but it seems to me that this could be a timinig issue caused by async client. Do we need to call `join()` to await completion of the API call before we proceed to the next - as described here? https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/asynchronous.html Thanks

Reply:
Hey @whummer. Actually this test was a previous one, I just removed the `putParameterReponse` variable assignment, as it wasn't used.
The test error is in `BasicFeaturesSDKV2Test.testGetSecretAsParam` because [this PR](https://github.com/localstack/localstack/pull/3263) hasn't been merged yet (which fixes the error that will make this test pass). Honestly, I created this draft PR just to reference in [this comment](https://github.com/localstack/localstack/pull/3263#issuecomment-757436150). As soon as I'm able to make a docker image build with changes I can complete the rests of the tests.
I'm under the impression that the flow is to first merge the changes in the localstack repo and then these tests will pass. Let me know if I got it wrong and what would be the right flow to correlate the changes and fixes from this repo and the localstack one.